int64 to
int64 from
---
int64 x_i
int64 y_i
int64 theta_i
int64 in_a_circle_i
int64 s_i
int64 g_i